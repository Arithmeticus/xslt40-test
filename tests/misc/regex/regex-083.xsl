<?xml version="1.0" encoding="UTF-8"?>
<t:transform xmlns:t="http://www.w3.org/1999/XSL/Transform" version="2.0">
<!-- Purpose: Testcase with xsl:analyze-string to test context item, position and size inside xsl:(non)-matching-substring.-->

   <t:output method="xml" encoding="UTF-8"/>

   <t:template match="/doc">
      <t:text>
</t:text>
      <out>
         <t:text>
</t:text>
         <E1>
            <t:analyze-string select="str[3]" regex="a">
    	          <t:matching-substring>
    		            <t:text>Matching-Item: </t:text>
                  <t:value-of select="current()"/>
    		            <t:text>
</t:text>
    		            <t:text>Position: </t:text>
                  <t:value-of select="position()"/>
    		            <t:text>
</t:text>
    		            <t:text>Size: </t:text>
                  <t:value-of select="last()"/>
    		            <t:text>
**
</t:text>
    	          </t:matching-substring>
    	          <t:non-matching-substring>
    		            <t:text>Non-matching-Item: </t:text>
                  <t:value-of select="current()"/>
    		            <t:text>
</t:text>
    		            <t:text>Position: </t:text>
                  <t:value-of select="position()"/>
    		            <t:text>
</t:text>
    		            <t:text>Size: </t:text>
                  <t:value-of select="last()"/>
    		            <t:text>
**
</t:text>
    	          </t:non-matching-substring>
            </t:analyze-string>
         </E1>
      </out>
   </t:template>
</t:transform>
