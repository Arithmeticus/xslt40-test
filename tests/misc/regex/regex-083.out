<?xml version="1.0" encoding="UTF-8"?>
<out>
<E1>Non-matching-Item: b
Position: 1
Size: 6
**
Matching-Item: a
Position: 2
Size: 6
**
Non-matching-Item: n
Position: 3
Size: 6
**
Matching-Item: a
Position: 4
Size: 6
**
Non-matching-Item: n
Position: 5
Size: 6
**
Matching-Item: a
Position: 6
Size: 6
**
</E1></out>